### PR TITLE
nixos/testing: restrict arguments to makeTest

### PR DESCRIPTION
**Pull request by ncfavier**
Disallow passing arbitrary arguments to `makeTest` since they are not used; this can help catch mistakes.

Then fix those mistakes, as well as some unrelated errors and warnings.

### CODE REVIEW

File: nixos/tests/without-nix.nix [hunk 1]
```diff
@@ -4,12 +4,6 @@ import ./make-test-python.nix ({ lib, ... }: {
     maintainers = [ ericson2314 ];
   };
 
-  nixpkgs.overlays = [
-    (self: super: {
-      nix = throw "don't want to use this";
-    })
-  ];
-
```
Comment (ncfavier) [inline nixos/tests/without-nix.nix:12]: @Ericson2314 note that the test fails if I add this to the actual machine configuration.

**Comment by roberth** [comment]
@ofborg test without-nix installer.simple

File: nixos/tests/without-nix.nix (hunk 1 above)
Comment (roberth) [reply nixos/tests/without-nix.nix:12]: Got nerd-sniped and fixed it.

File: nixos/tests/without-nix.nix [hunk 2]
```diff
@@ -4,14 +4,23 @@ import ./make-test-python.nix ({ lib, ... }: {
...
+        nix = throw "don't want to use pkgs.nix";
+        nixVersions = lib.mapAttrs (k: throw "don't want to use pkgs.nixVersions.${k}") super.nixVersions;
+        # aliases, some deprecated
+        nix_2_3 = throw "don't want to use pkgs.nix_2_3";
+        nix_2_4 = throw "don't want to use pkgs.nix_2_4";
+        nix_2_5 = throw "don't want to use pkgs.nix_2_5";
+        nix_2_6 = throw "don't want to use pkgs.nix_2_6";
+        nixFlakes = throw "don't want to use pkgs.nixFlakes";
+        nixStable = throw "don't want to use pkgs.nixFlakes";
+        nixUnstable = throw "don't want to use pkgs.nixFlakes";
```
Comment (ncfavier) [inline nixos/tests/without-nix.nix:20]: Mismatch intentional?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -33613,7 +33613,7 @@ with pkgs;
             then import test
             else test;
           calledTest = if lib.isFunction loadedTest
-            then callPackage loadedTest {}
+            then loadedTest { inherit pkgs lib; }
```
Comment (roberth) [inline pkgs/top-level/all-packages.nix:33616]: This was supposed to be similar to applying the args in `import ./make-test-python.nix ({ pkgs, ...}: `.
I've added some commits to fix this and test it properly.

File: nixos/tests/without-nix.nix (hunk 2 above)
Comment (roberth) [reply nixos/tests/without-nix.nix]: Good catch! Thanks :)

**Comment by ncfavier** [comment]
LGTM!

**Comment by roberth** [comment]
The chromium package wasn't buildable.

@ofborg test caddy ceph-multi-node cri-o

**Review by roberth — APPROVED** [review]

**Comment by roberth** [comment]
Awesome. Thank you @ncfavier !

**Pull request merged**